validateFileNames.js

```diff
@@ -0,0 +1,94 @@
+const shelljs = require('shelljs');
+
+// Regex that defines our file and directory naming convention
+const camelCasedFileNameRegex = /^[.]?([a-z])+([0-9]|[a-zA-Z]|[.])*$/;
+
+// Files and directories that are exempt from the naming convention
+const ignoredFilesAndDirectories = [
```

Comment:
I think we should use `minimatch` to match patterns of files. It works like this:

```
> minimatch('path/to/Dockerfile', 'Dockerfile', { matchBase: true })
true

> minimatch('customTypings/lodash-modules.d.ts', 'customTypings/*', { matchBase: true })
true

```

Using `minimatch` we can simplify `isIgnored`.